ibis/expr/types/relations.py

```diff
@@ -225,6 +225,26 @@ def cast(self, schema: SupportsSchema) -> Table:
             ...
         ibis.common.exceptions.IbisError: Cast schema has fields that are not in the table: ['foo']
         """
+        return self._cast(schema, strict=True)
+
+    def try_cast(self, schema: SupportsSchema, strict=True) -> Table:
+        """Cast the columns of a table.
+        If the cast fails for a row, the value is returned
+        as `NULL` or `NaN` depending on backend behavior.
+
+        Parameters
+        ----------
+        schema
+            Mapping, schema or iterable of pairs to use for casting
+
+        Returns
+        -------
+        Table
+            Casted table
+        """
+        return self._cast(schema, strict=False)
+
+    def _cast(self, schema: SupportsSchema, strict=True) -> Table:
```

Comment:
Could also do something like

```suggestion
    def _cast(self, schema: SupportsSchema, cast_method: str = "cast") -> Table:
```

to avoid boolean logic below. Either works though.

Reply:
that seems cleaner